scripts/strict_kernel_warn.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+module Kernel
```

Comment:
1. I might document this with a comment? Big change in behavior from typical Ruby
2. This seems like a `lib/core_ext` kinda file to me, I would move it there? Since lib is not autoloaded it should be safe

```suggestion
# Module that overrides Kernel#warn to immediately exit
# Should NOT be included in normal web server mode, is only used in specific scripts
module Kernel
```

Reply:
Yeah, good point on both counts. I'll make those updates.

Reply:
@lmgeorge What do you think of the updated approach? I think it's an improvement. I hadn't realized that these warnings were being emitted to stderr, so it becomes a test of "fail if there's any stderr".

Example build failure: https://gitlab.login.gov/lg/identity-idp/-/jobs/1515745

Example output:

```
--- rubocop ---
mkdir -p tmp
bundle exec rubocop --parallel --format progress --format junit --out rubocop.xml --display-only-failed --color 2> tmp/rubocop.txt
Inspecting 2259 files
.......
2259 files inspected, no offenses detected
[ -s tmp/rubocop.txt ] && (printf "Error: Unexpected stderr output from Rubocop\n"; cat tmp/rubocop.txt; exit 1)
Error: Unexpected stderr output from Rubocop
lib/identity_config.rb: Metrics/LineLength has the wrong namespace - replace it with Layout/LineLength
lib/identity_config.rb: Metrics/LineLength has the wrong namespace - replace it with Layout/LineLength
make: *** [Makefile:71: lint] Error 1
```

Reply:
I sort of lean towards @lmgeorge's concerns on this in that the change feels pretty heavy (though limited in scope) for the upside. The only alternative I have is to grep the output for the message we don't expect to see. It's not quite the same though. I don't intend for this to be a blocking suggestion since anything going sideways on this will hopefully be pretty apparent.

Reply:
@mitchellhenke I've since changed from the initial approach in recent commits.

Reply:
@aduth I can agree with this approach of grepping the stderr output. 

My main concern is that it can be easy to see an existing pattern and replicate it elsewhere even when strong language in the form of doc comments warn (heh) against it. I think this new approach addresses that concern. Thank you for being willing to flex on this!